core/common/src/main/java/alluxio/collections/UniqueFieldIndex.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.collections;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * An interface extending {@link FieldIndex}, represents a unique index. A unique index is an index
+ * where each index value only maps to one object.
+ *
+ * @param <T> type of objects in this {@link IndexedSet}
+ */
+@NotThreadSafe
```

Comment:
sorry for the confusion, I think this class is `ThreadSafe` as it is using `ConcurrentHashMap`